app/src/main/java/com/mensinator/app/settings/SettingsScreen.kt

```diff
@@ -126,6 +128,26 @@ fun SettingsScreen(
             },
             onOpenIntPicker = { viewModel.showIntPicker(it) }
         )
+        SettingText(
+            text = stringResource(StringSetting.PERIOD_NOTIFICATION_MESSAGE.stringResId),
+            onClick = { viewModel.showPeriodNotificationDialog(true) }
+        )
+
+        if (viewState.showPeriodNotificationDialog) {
+            val initMessage = viewState.periodNotificationMessage
```

Comment:
Imo it would be better to have this logic in the VM.
It is quite complex and it has to do with data handling, both things preferabbly done in the VM. Imagine we want to write a test case for this, it would be difficult, as the code is in the UI layer right now.

Also, it needs a short comment to explain *why* we are running this code.

Reply:
I agree it's quite complex but I was trying to replicate the symptom method as asked so I took this from the symtom screen file :P
If there's a way to consolidate them that would be good

Reply:
I've done something with the commits `80bc3b23..17c20c0f` which tries to put all the duplicate code and null handling into functions etc
I'm not really happy with it though because it just makes wrappers around existing functions instead of what you asked for
Might be another case where you need to show the best idea again sorry :p

Reply:
It's probably fine for now. I will maybe refactor it at some point.